pptx.go

```diff
@@ -0,0 +1,66 @@
+package docconv
+
+import (
+	"archive/zip"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+// ConvertPptx converts an MS PowerPoint pptx file to text.
+func ConvertPptx(r io.Reader) (string, map[string]string, error) {
+	var size int64
+
+	// Common case: if the reader is a file (or trivial wrapper), avoid
+	// loading it all into memory.
+	var ra io.ReaderAt
+	if f, ok := r.(interface {
+		io.ReaderAt
+		Stat() (os.FileInfo, error)
+	}); ok {
+		si, err := f.Stat()
+		if err != nil {
+			return "", nil, err
+		}
+		size = si.Size()
+		ra = f
+	} else {
+		b, err := ioutil.ReadAll(r)
+		if err != nil {
+			return "", nil, nil
+		}
+		size = int64(len(b))
+		ra = bytes.NewReader(b)
+	}
+
+	zr, err := zip.NewReader(ra, size)
+	if err != nil {
+		return "", nil, fmt.Errorf("error unzipping data: %v", err)
+	}
+
+	zipFiles := mapZipFiles(zr.File)
+
+	contentTypeDefinition, err := getContentTypeDefinition(zipFiles["[Content_Types].xml"])
+	if err != nil {
+		return "", nil, err
+	}
+
+	meta := make(map[string]string)
+	var textBody string
+	for _, override := range contentTypeDefinition.Overrides {
+		f := zipFiles[override.PartName]
+
+		switch {
+		case override.ContentType == "application/vnd.openxmlformats-officedocument.presentationml.slide+xml" ||
+			override.ContentType == "application/vnd.openxmlformats-officedocument.drawingml.diagramData+xml":
+			body, err := parseDocxText(f)
+			if err != nil {
+				return "", nil, err
```

Comment:
Instead of this maybe we should do:

```go
return "", nil, fmt.Errorf("could not parse override %d as docx: %v", i, err)
```

Otherwise it may be hard to work out what override is causing grief.

Reply:
@xiaoxin01 are you able to use the suggested error message so it's clear what override failed? Does that make sense to you?

Reply:
> @xiaoxin01 are you able to use the suggested error message so it's clear what override failed? Does that make sense to you?

I have changed to https://github.com/sajari/docconv/pull/71/commits/70aeacc08d9391513506c9ee9d4590307943c805#diff-d15dc33d4a0ceafc6a34ff5007ceeab4R59-R63